diff --git a/cmake/SGSPlatform.cmake b/cmake/SGSPlatform.cmake
index dc66c58..e58c667 100644
--- a/cmake/SGSPlatform.cmake
+++ b/cmake/SGSPlatform.cmake
@@ -172,6 +172,11 @@ function(sgs_find_host_compiler)
         set(compiler gcc)
         execute_process(COMMAND ${SGS_HOST_C_COMPILER} -dumpversion OUTPUT_VARIABLE GCC_VERSION)
         string(REGEX MATCHALL "[0-9]+" GCC_VERSION_COMPONENTS ${GCC_VERSION})
+        list(LENGTH GCC_VERSION_COMPONENTS GCC_VERSION_COMPONENTS_NB)
+        if(GCC_VERSION_COMPONENTS_NB LESS "2")
+          execute_process(COMMAND ${SGS_HOST_C_COMPILER} --version OUTPUT_VARIABLE GCC_VERSION)
+          string(REGEX MATCHALL "[0-9]+" GCC_VERSION_COMPONENTS ${GCC_VERSION})  
+        endif()
         list(GET GCC_VERSION_COMPONENTS 0 GCC_MAJOR)
         list(GET GCC_VERSION_COMPONENTS 1 GCC_MINOR)
         set(cvers ${GCC_MAJOR}${GCC_MINOR})

