diff --git a/ElementsKernel/src/Lib/ModuleInfo.cpp b/ElementsKernel/src/Lib/ModuleInfo.cpp
index e32499e6..be87c6ec 100644
--- a/ElementsKernel/src/Lib/ModuleInfo.cpp
+++ b/ElementsKernel/src/Lib/ModuleInfo.cpp
@@ -247,11 +247,10 @@ const vector<string> linkedModules() {
 Path::Item getExecutablePath() {
 
 #ifdef __APPLE__
-  path self_proc {};
   char pathbuf[PATH_MAX + 1];
   unsigned int  bufsize = sizeof(pathbuf);
   _NSGetExecutablePath(pathbuf, &bufsize);
-  path self_exe = path(string(pathbuf));
+  Path::Item self_exe = Path::Item(string(pathbuf));
 #else
 
   Path::Item self_exe = getSelfProc() / "exe";
